tests/sdl2_glalphatest.c

```diff
@@ -110,7 +110,5 @@ int main(int argc, char* argv[]) {
 
   glDeleteTextures(1, &texture);
 
-  SDL_Quit();
```

Comment:
Otherwise all windows get destroyed and the tests fail because the canvas is gone (well, 0x0).

Reply:
So have these tests been failing forever?

Can we perhaps land all these test fixes separately (first), since they seem independent of the SDL2 update?

Reply:
I think these were passing with the old version, so something may have changed there. I can split the test fixes out though.

(The testfullscreen one has likely been failing for a while.)

Reply:
I did slightly worry that we have a behavioral change here?  My understanding is that switching to upstream SDL would essentially be a no-op but this seem to contradict that.

Can you explain a little more what you mean by "tests fail because the canvas is gone (well, 0x0)."?  Is this test run as part of CI?  i.e. would reverting this part of the change cause CI to fail?

Reply:
The tests pass with main+version_24 and start to fail after switching to a newer SDL (this update is by four releases).

I did some more tests and it seems the behaviour changes in 2.0.14, likely this commit: https://github.com/libsdl-org/SDL/commit/694fea8ad8fd074fae1ec30fe85267b6dc476f49. So before SDL wasn't shutting down the video subsystem and now it is.

(Error for the failing tests would be "Assertion failed: Invalid width: 0, should be 640")

Reply:
Sounds like something its probably worth fixing I guess?   Isn't it perfectly valid to call SDL_Quit here?  Or is this wrong in some way?

Reply:
The tests are relying on the "window" still existing after exiting to verify the image though... which only worked because `SDL_Quit` was broken and did nothing. (Quitting the video subsystem destroys all windows)

(`SDL_DestroyWindow` has always either cleared or resized the canvas in the emscripten backend, it might even remove the canvas in future if I ever get back to the multiple-window/canvas patches...)

Reply:
I see, thanks for clarifying.   Makes sense.   Could you replace these with a comment maybe `// Exiting without calling SDL_Quit so the test framework can the canvas data`  (or something like that.. I'm not sure I have the working corret.

Reply:
Ah, some of the other tests already have something like that:
https://github.com/emscripten-core/emscripten/blob/df034c938c8bdbb3b0a272e9794084e9bce8567c/tests/sdl2glshader.c#L169

Must have been missed in these ones due to the missing `SDL_Init` and the bug in quit... will update

Reply:
Maybe also a good idea to add a changlog entry for this? Both in general for the change to use latest upstream, and specifically for this noticeable issue.

Reply:
Ooh, yeah +4 SDL releases is probably worth a changelog entry! I'll try to come up with something...

(note that the issue here is only noticeable if you _don't_ call `SDL_Init` but do call `SDL_Quit`... otherwise it happened a few updates ago)

Reply:
I see, thanks! Yeah, I guess the breaking change here is not that important to mention then. But the +4 versions is :smile: Any positive highlights that people might notice could also be worth mentioning briefly in the log.